### PR TITLE
Added nicer messages for unverified email actions

### PR DESCRIPTION
**Pull request by will-sonnex**

### CODE REVIEW

File: packages/backend/package.json [hunk 1]
```diff
@@ -57,7 +57,7 @@
     "@typescript-eslint/eslint-plugin": "^5.0.0",
     "@typescript-eslint/parser": "^5.0.0",
     "aws-sdk": "^2.1295.0",
-    "better-sqlite3": "^8.0.1",
+    "better-sqlite3": "^11.3.0",
```
Comment (stabback) [inline packages/backend/package.json:60]: No change needed over 3 major versions?

File: packages/frontend/src/common/api-error-handling.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { ResponseError } from "../api";
+
+export const NO_CODE_GIVEN = "NONE";
+export const EMAIL_UNVERIFIED = "EMAIL_UNVERIFIED";
```
Comment (stabback) [inline packages/frontend/src/common/api-error-handling.ts]: i'm a lover of SCREAMING_SNAKE_CASE constants

File: packages/frontend/src/common/api-error-handling.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+};
+
+export const isEmailUnverifiedError = async (
+  error: ResponseError
+): Promise<boolean> => {
+  return (await getErrorCode(error)) === EMAIL_UNVERIFIED;
+};
+
+export const EMAIL_UNVERIFIED_MESSAGE =
+  "Please verify your email address, log out, and log back in. Sorry.";
```
Comment (stabback) [inline packages/frontend/src/common/api-error-handling.ts]: 🇨🇦

File: packages/frontend/src/pages/EditOpportunity/EditOpportunity.tsx [hunk 4]
```diff
@@ -35,9 +43,15 @@ export const EditOpportunityPage = () => {
...
-      });
+      if (e instanceof ResponseError && (await isEmailUnverifiedError(e))) {
+        enqueueSnackbar(EMAIL_UNVERIFIED_MESSAGE, {
+          variant: "error",
+        });
+      } else {
+        enqueueSnackbar("Error editing opportunity.  Try again later.", {
+          variant: "error",
+        });
+      }
```
Comment (stabback) [inline packages/frontend/src/pages/EditOpportunity/EditOpportunity.tsx]: Feels like we should wrap this in a helper given the repetition

File: packages/frontend/src/index.tsx [hunk 5]
```diff
@@ -18,7 +18,7 @@ root.render(
       <UserProvider>
         <ApiProvider>
           <LocalizationProvider dateAdapter={AdapterLuxon}>
-            <SnackbarProvider maxSnack={3}>
+            <SnackbarProvider maxSnack={3} autoHideDuration={10000}>
```
Comment (stabback) [inline packages/frontend/src/index.tsx]: Unless they've asked for this I'd revert this, autohiding notifications is a bit of an accessibility rabbit hole.  You've gotta ensure all messages (of all supported lengths) have sufficient time on screen to be read by people of all abilities.

**Review by stabback — APPROVED** [review]
here are a few thoughts
some have meaning, some less so
all code is fleeting

File: packages/backend/package.json (hunk 1 above)
Comment (will-sonnex) [reply packages/backend/package.json:60]: Sqlite's interface is SQL. Doesn't change much.

**Pull request merged**